posts/hello-world-in-kafka-using-python.md

```diff
@@ -0,0 +1,171 @@
+
+![Simple Architecture](../images/kafka-image-two.jpg)
+
+This might look simple, but you're not finished. There are multiple challenges that can arise. 
+
+1. Events like clicks, requests, impressions and searches results in high frequency interaction/requests or data flow to your webserver and your primary database may not be equipped to scale seamlessly. This could introduce high latencies as more and more events pour in to the server.
+2. If you go ahead and store high frequency data in database systems like SQL or Mongo, it would be hard to introduce and reconstruct a new system or a database on all of the historical data. You lose the flexibility to extend the capabilities of your system by introducing new techologies.
+3. What if you have data processing systems in place to process these events to gain deeper insights? Since these systems wouldn't be capable of handling high frequency reads and you wouldn't have access to the true source of data it is practically impossible to experiment with various data processing, machine learning algorithms on all of the data.
+3. Each application can follow their own data formats, which means that you will need systems for data transformations when there is exchange of data across these applications.
+
+All these problems can be better addressed by bringing a streaming platform like Kafka into the picture. A streaming platform is a system that can perform the following:
+
+1. Store huge amount of data that can be persistent, checksummed and replicated for fault tolerance 
+2. Process continuous flow of data (data streams) in real time across systems
+3. Allow applications to publish data or data streams independently and agnostic to the application / service consuming it
+
+> Interesting! How different or similar is it from traditional databases?
+
+Although Kafka can store persistent data, it is NOT a database.
+
+Kafka not only allows applications to push or pull continuous flow of data, but also deals with processing them to build and support real time applications. This is different than performing CRUD operations on passive data or running queries on traditional databases. 
+
+> That sounds convincing! But how does Kafka solve the above mentioned challenges and why would one need a system like this?
+
+Kafka is a distributed platform and built for scale, which means it can handle sky-high frequency reads and writes & store huge volumes of data. It ensures that the data is always reliable. It also supports strong mechanisms for recovery from failures. Here are some of key aspects on why one should be using Kafka:
+
+### 1. Simplify the backend architecture
+
+Look at how a complex architecture can be simplified and streamlined with the help of Kafka
+
+![Simple Architecture](../images/kafka-image-three.jpg)
```

Comment:
I think putting arrows in both directions from all the items would show that any of the items can be a producer/consumer